app/services/pipeline_report_service.rb

```diff
@@ -474,22 +474,23 @@ def find_taxa_to_highlight(sorted_genus_tax_ids, counts_by_tax_level)
     return highlighted_tax_ids
   end
 
-  def tag_pathogens(counts_by_tax_level, lineage_by_tax_id)
-    counts_by_tax_level[TaxonCount::TAX_LEVEL_SPECIES].each do |species_tax_id, species_info|
-      lineage = lineage_by_tax_id[species_tax_id] ? lineage_by_tax_id[species_tax_id] : lineage_by_tax_id[species_info[:genus_tax_id]]
-      name_columns = lineage.keys.select { |cn| cn.include?("_name") }
+  def tag_pathogens(counts_by_tax_level)
+    counts_by_tax_level[TaxonCount::TAX_LEVEL_SPECIES].each do |_species_tax_id, species_info|
       pathogen_tags = []
       TaxonLineage::PRIORITY_PATHOGENS.each do |category, pathogen_list|
```

Comment:
We should make a task of moving this data structure to a hash keyed by the name...

Reply:
Do you mean restructuring `counts_by_tax_level` to look like {"species": {...}, "genus": {...}} rather than {1: {...}, 2: {...}}? I'll create a task/add a TODO comment.

Reply:
You should do that too! 
But I actually meant the `TaxonLineage::PRIORITY_PATHOGENS` should be refactored to hash of `{<name_or_tax_id>: <pathogen_tag>}`.
Sorry, should have been more clear...